src/molgenis/eucan_connect/mica.py

```diff
@@ -0,0 +1,761 @@
+        self.studyMrefs = {}
+        self.warnings: List[EucanWarning] = []
+
+    def mica_data(self, catalogue: Catalogue) -> CatalogueData:
+        """
+        Retrieves data from the provided source catalogue
+        Data categories are: dataset, network, study and variable.
+
+        The study data category contains, besides study information, also
+        person, data collection events and population data.
+        For the conversion to the EUCAN-Connect catalogue only data
+        from the study category is selected as the other three data categories
+        can not be mapped to the EUCAN-Connect Catalogue data model.
+        """
+
+        self.catalogue = catalogue
+        self.printer.print(f"🗑 Get {self.catalogue.description} studies")
+
+        # First retrieve the list with the studies in the source catalogue:
+        mica_studies = self.get_mica_study_ids()
+        print("aantal studies", len(mica_studies))
+
+        # Secondly retrieve the full study data
+        mica_study_data = self.get_mica_study_data(mica_studies)
+
+        if len(mica_studies) != len(mica_study_data):
+            # raise EucanError
+            warning = EucanWarning(
+                f"Number of studies with data ({len(mica_study_data)}) "
+                f"does not equal the total number of studies ({len(mica_studies)})"
+            )
+            self.printer.print_warning(warning)
+            self.warnings.append(warning)
+
+        # Convert the mica study data to the EUCAN-Connect Catalogue model:
+        catalogue_data = self._convert_studies_to_eucan(mica_study_data)
+
+        # Add the study mrefs to the data
+        self.printer.print(
+            "Adding the event, person and population mrefs to the studies"
+        )
+        catalogue_data = self.eucan_session.set_study_mrefs(
+            catalogue_data, self.studyMrefs
+        )
+
+        return catalogue_data
+
+    def get_mica_study_ids(self, start: int = 0, batch_size: int = 100) -> List:
+        """
+        Get a list with study IDs from a catalogue by using the Mica REST API Client.
+        The REST API Client that is used here returns not only the IDs,
+        but also study information. It however turns out that this is
+        'summary' information and it does not contain all necessary data
+        for the EUCAN-Connect Catalogue. Therefore this API is only used for
+        retrieving a list with study IDs. Depending on the value of param amount
+        only the total amount of studies or the real study information is returned
+        @param start: study number (row) to start from, default is 0
+        @param batch_size: number of studies to be returned
+                           if != totalCount, a loop should be created
+        """
+        studies = []
+        study_url = self.catalogue.catalogue_url + "/ws/studies/_rql"
+        study_query = (
+            "query=study(limit(" + str(start) + "," + str(batch_size) + "),fields(*))"
```

Comment:
You can use string formatting here, which reads a bit nicer:

```suggestion
            f"query=study(limit({str(start)},{str(batch_size)}"),fields(*))"
```

Reply:
Implemented